### PR TITLE
Improve logging in BasicMapTest

### PR DESCRIPTION
**Pull request by vbekiaris**
Add failure message that indicates in which step the thread that tries to put stopped making progress (for #5364 ). Also, if `tryPut` times out, fail the test immediately instead of waiting for timeout.

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java [hunk 1]
```diff
@@ -742,8 +748,17 @@ public void run() {
         assertOpenEventually(waitTryPutFailure);
 
         map.unlock("key1");
-        assertOpenEventually(latch);
+        try {
+            boolean completed = latch.await(ASSERT_TRUE_EVENTUALLY_TIMEOUT, TimeUnit.SECONDS);
+            assertTrue("tryPut operations were not completed in time, counter was " + counter.get() + " and failure"
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java]: Unwrapped `assertOpenEventually` to evaluate `counter.get()` and `failureMessageBuilder.toString()` at time of (potential) failure.

Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java]: I'm never crazy about doing asserts on string content due to fragility.

**Comment by pveentjer** [comment]
:+1:

File: hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java (hunk 1 above)
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java]: This is actually `assertTrue(longFailureMessage, booleanToAssert)`. Put the blame on the new side-by-side github review layout :-)

**Pull request merged**